lib/va_profile/profile_information/transaction_response.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require 'va_profile/models/transaction'
+require 'va_profile/response'
+
+module VAProfile
+  module ProfileInformation
+    class TransactionResponse < VAProfile::Response
+      extend SentryLogging
+
+      attribute :transaction, VAProfile::Models::Transaction
+      ERROR_STATUS = 'COMPLETED_FAILURE'
+
+      attr_reader :response_body
+
+      def self.from(raw_response = nil, _user = nil)
+        @response_body = raw_response&.body
+        if error?
+          log_message_to_sentry(
+            'VAProfile transaction error',
+            :error,
+            { response_body: @response_body },
+            error: :va_profile
+          )
+        end
+
+        new(
+          raw_response&.status,
+          transaction: VAProfile::Models::Transaction.build_from(@response_body)
+        )
+      end
+
+      def self.error?
+        @response_body.try(:[], 'tx_status') == ERROR_STATUS
+      end
+    end
+
+    class AddressTransactionResponse < TransactionResponse
+      attribute :response_body, String
+
+      def self.from(*args)
+        return_val = super
+
+        log_error
+
+        return_val.response_body = @response_body
+        return_val
+      end
+
+      def changed_field
+        return :address unless response_body['tx_output']
```

Comment:
You could put `response_body['tx_output']` into a variable so it's not referenced multiple times